pkg/azureconstants/azure_constants.go

```diff
@@ -81,6 +81,7 @@ const (
 	UserAgentField                = "useragent"
 	VolumeAttributePartition      = "partition"
 	WellKnownTopologyKey          = "topology.kubernetes.io/zone"
+	AzureDiskCSITopologyKey       = "topology.disk.csi.azure.com"
```

Comment:
The actual topology key used by the driver is dependent on the driver's name and is initialized here: https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/840111893e6efd8b0890af212e14d456d157ce01/pkg/azuredisk/azuredisk_v2.go#L133. When installing the V2 driver SxS, the topology key will be `topology.disk2.csi.azure.com/zone`

Reply:
Added another constant for topology.disk2.csi.azure.com/zone

Reply:
We cannot rely on constant values for the driver-specific topology since it is derived from the driver name which is passed on the command line to the driver. We'll need to plumb the generated value through to make it available to the `controller` package. Adding a `topologyKey` field to `controller.SharedState would be a good way to do it.

Reply:
Yeah, that makes sense.

Reply:
Added topology key to the controller shared state and removed it from constants.